neonvm/controllers/webhook.go

```diff
@@ -0,0 +1,129 @@
+package controllers
+
+// Wrapper around the default VirtualMachine/VirtualMachineMigration webhook interfaces so that the
+// controller has a bit more control over them, without needing to actually implement that control
+// inside of the apis package.
+
+import (
+	"context"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+
+	vmv1 "github.com/neondatabase/autoscaling/neonvm/apis/neonvm/v1"
+)
+
+func validateUpdate(
+	cfg *ReconcilerConfig,
+	recorder record.EventRecorder,
+	oldObj runtime.Object,
+	newObj interface {
+		webhook.Validator
+		metav1.Object
+	},
+) error {
+	namespacedName := types.NamespacedName{
+		Namespace: newObj.GetNamespace(),
+		Name:      newObj.GetName(),
+	}
```

Comment:
Note to self: Should probably use `client.GetObjectKeyFromObject` instead of filling out the `NamespacedName` manually